src/fcntl/statx.rs

```diff
@@ -0,0 +1,371 @@
+
+    /// Retrieve number of hard links, if it has been returned by kernel.
+    pub fn nlinks(&self) -> Option<u32> {
+        if Mask::STATX_NLINK.bits() & self.inner.stx_mask > 0 {
+            Some(self.inner.stx_nlink)
+        } else {
+            None
+        }
+    }
+
+    /// Retrieve uid (owner, user ID), if it has been returned by kernel.
+    pub fn uid(&self) -> Option<Uid> {
+        if Mask::STATX_NLINK.bits() & self.inner.stx_mask > 0 {
+            Some(Uid::from_raw(self.inner.stx_uid))
+        } else {
+            None
+        }
+    }
+
+    /// Retrieve gid (group ID), if it has been returned by kernel.
+    pub fn gid(&self) -> Option<Gid> {
+        if Mask::STATX_NLINK.bits() & self.inner.stx_mask > 0 {
+            Some(Gid::from_raw(self.inner.stx_uid))
+        } else {
+            None
+        }
+    }
+
+    /// Retrieve file access time, if it has been returned by kernel
+    pub fn atime(&self) -> Option<libc::statx_timestamp> {
+        if Mask::STATX_ATIME.bits() & self.inner.stx_mask > 0 {
+            Some(self.inner.stx_atime)
+        } else {
+            None
+        }
+    }
+
+    /// Retrieve file modification time, if it has been returned by kernel
+    pub fn mtime(&self) -> Option<libc::statx_timestamp> {
+        if Mask::STATX_MTIME.bits() & self.inner.stx_mask > 0 {
+            Some(self.inner.stx_mtime)
+        } else {
+            None
+        }
+    }
+
+    /// Retrieve file attribute change time, if it has been returned by kernel
+    pub fn ctime(&self) -> Option<libc::statx_timestamp> {
+        if Mask::STATX_CTIME.bits() & self.inner.stx_mask > 0 {
+            Some(self.inner.stx_ctime)
+        } else {
+            None
+        }
+    }
+
+    /// Retrieve file birth (creation) time, if it has been returned by kernel
+    pub fn btime(&self) -> Option<libc::statx_timestamp> {
+        if Mask::STATX_BTIME.bits() & self.inner.stx_mask > 0 {
+            Some(self.inner.stx_btime)
+        } else {
+            None
+        }
+    }
+
+    /// Retrieve inode number, if it has been returned by kernel
+    pub fn ino(&self) -> Option<u64> {
+        if Mask::STATX_INO.bits() & self.inner.stx_mask > 0 {
+            Some(self.inner.stx_ino)
+        } else {
+            None
+        }
+    }
+
+    /// Retrieve file size, in bytes, if it has been returned by kernel
+    pub fn size(&self) -> Option<u64> {
+        if Mask::STATX_SIZE.bits() & self.inner.stx_mask > 0 {
+            Some(self.inner.stx_size)
+        } else {
+            None
+        }
+    }
+
+    /// Retrieve file size as a number of blocks, if it has been returned by kernel
+    pub fn blocks(&self) -> Option<u64> {
+        if Mask::STATX_BLOCKS.bits() & self.inner.stx_mask > 0 {
+            Some(self.inner.stx_blocks)
+        } else {
+            None
+        }
+    }
+
+    /// Retrieve file size as a number of blocks, if it has been returned by kernel
+    pub fn blksize(&self) -> Option<u64> {
```

Comment:
Is this identical to `blocks`?

Reply:
No, probably I just forgot to fill in new doccomment and implementation after copying neighbouring method.

Reply:
What is it supposed to do then?  Because it's just returning the same stx_blocks variable.

Reply:
Fixed to return `blocksize`, fixed the doccomment.